### PR TITLE
Do not delay `Database_Upgrader` service

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Improvement to https://github.com/google/web-stories-wp/pull/8792. Do not delay the service and hook into admin_init on register.

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [ ] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [ ] I have added a matching `Type: XYZ` label to the PR

---

Fixes #

### CODE REVIEW

File: includes/Database_Upgrader.php [hunk 1]
```diff
@@ -104,20 +104,7 @@ public function __construct( Injector $injector ) {
...
-	/**
-	 * Get the action priority to use for registering the service.
-	 *
-	 * @since 1.6.0
-	 *
-	 * @return int Registration action priority to use.
-	 */
-	public static function get_registration_action_priority(): int {
-		return 5;
+		add_action( 'admin_init', [ $this, 'run_upgrades' ], 5 );
```
Comment (swissspidy) [inline includes/Database_Upgrader.php:107]: Is there still a good reason for priority 5?

File: includes/Database_Upgrader.php [hunk 2]
```diff
@@ -104,20 +104,7 @@ public function __construct( Injector $injector ) {
 	 * @return void
 	 */
 	public function register() {
```
Comment (swissspidy) [inline includes/Database_Upgrader.php:106]: Note that tests need updating now and are failing.

File: includes/Database_Upgrader.php (hunk 1 above)
Comment (spacedmonkey) [reply includes/Database_Upgrader.php:107]: It used to be set to 5.

Comment (swissspidy) [reply includes/Database_Upgrader.php:107]: I know, but is it worth reconsidering this?

With #8513 we have other ways to ensure that the database upgrade runs before anything else.

Comment (spacedmonkey) [reply includes/Database_Upgrader.php:107]: More than happy to have to default of 10.

File: includes/Database_Upgrader.php (hunk 2 above)
Comment (spacedmonkey) [reply includes/Database_Upgrader.php:106]: Tests passing.

**Review by swissspidy — APPROVED** [review]

**Pull request merged**